core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -107,7 +108,13 @@ private String metadataFileLocation(Table table) {
 
   private FileIO fileIO(Table table) {
     if (table.io() instanceof HadoopConfigurable) {
-      ((HadoopConfigurable) table.io()).serializeConfWith(conf -> new SerializableConfiguration(conf)::get);
+      HadoopConfigurable configurableIo = (HadoopConfigurable) table.io();
+      if ("true".equalsIgnoreCase(configurableIo.getConf().get(ConfigProperties.CONFIG_SERIALIZATION_DISABLED))) {
```

Comment:
I think we should use: `((HadoopConfigurable) table.io()).getConf().getBoolean()`

Reply:
Yep, good idea